packages/assets-controllers/src/TokenRatesController.test.ts

```diff
@@ -749,335 +622,362 @@ describe('TokenRatesController', () => {
+              }),
+            },
+          },
+          async ({ controller, triggerNetworkStateChange }) => {
+            const updateExchangeRatesSpy = jest
+              .spyOn(controller, 'updateExchangeRates')
+              .mockResolvedValue();
+            triggerNetworkStateChange({
+              ...defaultNetworkState,
+              selectedNetworkClientId: defaultSelectedNetworkClientId,
+            });
 
-        expect(updateExchangeRatesSpy).not.toHaveBeenCalled();
+            expect(updateExchangeRatesSpy).not.toHaveBeenCalled();
+          },
+        );
       });
 
       it('should not update exchange rates when chain ID changes', async () => {
-        const getNetworkClientById = buildMockGetNetworkClientById({
-          'AAAA-BBBB-CCCC-DDDD': buildCustomNetworkClientConfiguration({
-            chainId: toHex(1338),
-            ticker: 'TEST',
-          }),
-        });
-        let networkStateChangeListener: (state: NetworkState) => Promise<void>;
-        const onNetworkStateChange = jest
-          .fn()
-          .mockImplementation((listener) => {
-            networkStateChangeListener = listener;
-          });
-        const controller = new TokenRatesController({
-          interval: 100,
-          getNetworkClientById,
-          chainId: toHex(1337),
-          ticker: 'TEST',
-          selectedAddress: defaultSelectedAddress,
-          onPreferencesStateChange: jest.fn(),
-          onTokensStateChange: jest.fn(),
-          onNetworkStateChange,
-          tokenPricesService: buildMockTokenPricesService(),
-        });
-        const updateExchangeRatesSpy = jest
-          .spyOn(controller, 'updateExchangeRates')
-          .mockResolvedValue();
-
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        await networkStateChangeListener!({
-          ...defaultNetworkState,
-          selectedNetworkClientId: 'AAAA-BBBB-CCCC-DDDD',
-        });
+        await withController(
+          {
+            options: {
+              interval: 100,
+            },
+            mockNetworkClientConfigurationsByNetworkClientId: {
+              'AAAA-BBBB-CCCC-DDDD': buildCustomNetworkClientConfiguration({
+                chainId: toHex(1338),
+                ticker: 'TEST',
+              }),
+            },
+          },
+          async ({ controller, triggerNetworkStateChange }) => {
+            const updateExchangeRatesSpy = jest
+              .spyOn(controller, 'updateExchangeRates')
+              .mockResolvedValue();
+            triggerNetworkStateChange({
+              ...defaultNetworkState,
+              selectedNetworkClientId: defaultSelectedNetworkClientId,
+            });
 
-        expect(updateExchangeRatesSpy).not.toHaveBeenCalled();
+            expect(updateExchangeRatesSpy).not.toHaveBeenCalled();
+          },
+        );
       });
 
       it('should clear contractExchangeRates state when ticker changes', async () => {
```

Comment:
Hmm, looks like this test name wasn't changed when the state property was changed either. Should we fix that in this PR too?